libraries/azqueue/test/db/versions/0001.yml

```diff
@@ -0,0 +1,153 @@
+        where msgs.queue_name = azure_queue_count.queue_name);
+      end
+
+  azure_queue_put:
+    description: |
+      Put the given message into the given queue.  The message will not be visible until
+      after the visible timestamp, and will disappear after the expires timestamp.
+    mode: write
+    serviceName: test-azqueue
+    args: queue_name text, message_text text, visible timestamp, expires timestamp
+    returns: void
+    body: |-
+      begin
+        insert into azure_queue_messages (
+            queue_name,
+            message_id,
+            message_text,
+            inserted,
+            visible,
+            expires
+          ) values (
+            azure_queue_put.queue_name,
+            public.gen_random_uuid(),
+            azure_queue_put.message_text,
+            now(),
+            azure_queue_put.visible,
+            azure_queue_put.expires
+          );
+        execute 'notify ' || quote_ident(queue_name);
+      end
+
+  azure_queue_get:
+    description: |
+      Get up to `count` messages from the given queue, setting the `visible`
+      column of each to the given value.  Returns a `message_id` and
+      `pop_receipt` for each one, for use with `azure_queue_delete` and
+      `azure_queue_update`.
+    mode: write
+    serviceName: test-azqueue
+    args: queue_name text, visible timestamp, count integer
+    returns: table (message_id uuid, message_text text, pop_receipt uuid)
+    # This uses the queueing logic described many places around the web but
+    # best in https://tnishimura.github.io/articles/queues-in-postgresql/
+    #
+    # The magic sauce is
+    #
+    #   for update -- locks returned rows
+    #   skip locked -- skips locked rows
+    #
+    # The first ensures that no two concurrent transactions return the
+    # same message, while the second ensures that concurrent transactions
+    # do not serialize themselves but instead immediately return distinct
+    # messages, if they are available.
+    body: |-
+      begin
+        raise info 'oh dear %s', azure_queue_get.visible::text;
+        return query update azure_queue_messages m1
+          set
+            pop_receipt = public.gen_random_uuid(),
+            visible = azure_queue_get.visible
+          where
+            m1.message_id in (
+              select m2.message_id from azure_queue_messages m2
+              where m2.queue_name = azure_queue_get.queue_name
+                and m2.visible < now()
+                and m2.expires > now()
+              order by m2.inserted
+              for update skip locked
+              limit count
+          )
+          returning m1.message_id, m1.message_text, m1.pop_receipt;
+      end
+
+  azure_queue_delete:
+    description: |
+      Delete the message identified by the given `queue_name`, `message_id` and
+      `pop_receipt`.
+    mode: write
+    serviceName: test-azqueue
+    args: queue_name text, message_id uuid, pop_receipt uuid
```

Comment:
Silly question but why do we need anything other than `message_id`

Reply:
True, since that's the primary key.  This is reflecting the existing behavior of the Azure Queues API.  In that context, `pop_receipt` is a bearer proof that the caller is the one who popped the message from the queue, and I suppose queue_name helps to direct the request to the appropriate server.  In the postgres context, both function as secondary checks after the primary key index is used to find the row.